packages/sdk-react/src/components/receivables/InvoiceDetails/CreateReceivable/sections/components/ReminderSection/hooks/useValidateCounterpart.ts

```diff
@@ -0,0 +1,55 @@
+import { useCallback, useEffect, useState } from 'react';
+import { useFormContext } from 'react-hook-form';
+
+import { isIndividualCounterpart } from '@/components/counterparts/helpers';
+import { useMoniteContext } from '@/core/context/MoniteContext';
+import { useCounterpartById } from '@/core/queries';
+
+interface UseValidateCounterpartResult {
+  isEmailValid: boolean;
+  areRemindersEnabled: boolean;
+}
+
+export const useValidateCounterpart = (): UseValidateCounterpartResult => {
+  const { getValues } = useFormContext();
+  const { api } = useMoniteContext();
+
+  const [isEmailValid, setIsEmailValid] = useState(false);
+  const [areRemindersEnabled, setAreRemindersEnabled] = useState(true);
+
+  const counterpartId = getValues('counterpart_id'); //ToDo: Adjust this to match the field name of counterpart Form
+
+  const counterpartQuery = useCounterpartById(counterpartId);
+  const { data: counterpart } = counterpartQuery;
+
+  const validateEmailAndReminders = useCallback(async () => {
+    if (!counterpartId || !counterpartQuery.data || !counterpart) return;
+
+    let counterpartEmail: string | undefined;
```

Comment:
In general, the use of `let` in Monite is considered legal. Yes, we are aware of initiatives to drop `const`, but in our opinion, using `const` alone is more beneficial than problematic. Moreover, `const` is more in line with the declarative programming ideology, where the whole types tree can be inferred from the data. So, I would refactor usage if let with something like:

```tsx
const counterpartEmail = counterpartQuery.data.type === 'individual'
      ? counterpartQuery.data.individual.email
      : counterpartContacts.find((contact) => contact.is_default)?.email
```

Reply:
Changed from let to const